api_app/db/repositories/shared_services.py

```diff
@@ -26,6 +27,10 @@ def shared_service_query(shared_service_id: str):
     def active_shared_services_query():
         return f'SELECT * FROM c WHERE {IS_ACTIVE_CLAUSE} AND c.resourceType = "{ResourceType.SharedService}"'
 
+    @staticmethod
+    def active_shared_service_with_template_name_query(template_name: str):
+        return f'SELECT * FROM c WHERE {IS_ACTIVE_CLAUSE} AND c.resourceType = "{ResourceType.SharedService}" AND c.templateName = "{template_name}"'
```

Comment:
Do we need the template version in here too?

Reply:
This is a great question that I should've clarified explicitly :) 

If we had a template version here, that would mean that we allow several active shared services of the same resource type, but of different version.

Currently, I think we don't want to allow that (cc @damoodamoo  @marrobi), and instead we want to implement an upgrade action, which would upgrade an already existing shared service to a new version (see https://github.com/microsoft/AzureTRE/issues/141). 

So, currently, once a shared service is deployed, it cannot then be upgraded to a new version. This limitation will be solved as part of that issue I linked above